File: tenis-quarkus/src/main/java/br/unitins/joaovittor/basqueteiros/Marca/service/MarcaService.java
package br.unitins.joaovittor.basqueteiros.Marca.service;

import java.util.List;

import br.unitins.joaovittor.basqueteiros.Marca.dto.MarcaDTO;
import br.unitins.joaovittor.basqueteiros.Marca.dto.MarcaResponseDTO;
import jakarta.validation.Valid;

public interface MarcaService {
    
    public MarcaResponseDTO create(@Valid MarcaDTO dto);
    public void update(Long id, MarcaDTO dto);
    public boolean delete(Long id);
    public MarcaResponseDTO findById(Long id);
    public List<MarcaResponseDTO> findByNome(String nome);
    public List<MarcaResponseDTO> findAll(int page, int pageSize);
    public List<MarcaResponseDTO> findAll();
    public List<MarcaResponseDTO> findByNome(int page, int pageSize, String nome);
    public long count();
}

File: tenis-angular/src/app/services/marca.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Marca } from '../models/marca.model';

@Injectable({
  providedIn: 'root'
})
export class MarcaService {
  private baseUrl = 'http://localhost:8080/marcas';

  constructor(private httpClient : HttpClient){

  }

  findAll(page?: number, pageSize?: number): Observable<Marca[]> {
    let params = {};
    if (page !== undefined && pageSize !== undefined) {
      params = {
        page: page.toString(),
        pageSize: pageSize.toString()
      }
    }
    return this.httpClient.get<Marca[]>(this.baseUrl, {params}); 
  }
  count(): Observable<number> {
    return this.httpClient.get<number>(`${this.baseUrl}/count`); 
  }

  findById(id: string): Observable<Marca> {
    return this.httpClient.get<Marca>(`${this.baseUrl}/search/id/${id}`); 
  }

  insert(marca: Marca): Observable<Marca> {
    return this.httpClient.post<Marca>(this.baseUrl, marca);
  }

  update(marca: Marca): Observable<Marca> {
    const data = {
      nome: marca.nome,
      logo: marca.logo,
    }
    return this.httpClient.put<any>(`${this.baseUrl}/${marca.id}`, data); 
  }

  delete(marca: Marca): Observable<any>{
    return this.httpClient.delete<any>(`${this.baseUrl}/${marca.id}`); 
  }
}
